src/find/matchers/size.rs

```diff
@@ -0,0 +1,174 @@
+use std::error::Error;
+use std::io::{stderr, Write};
+use walkdir::DirEntry;
+
+use find::matchers::{ComparableValue, Matcher, MatcherIO};
+
+#[derive(Clone, Copy, Debug)]
+enum Unit {
+    Byte,
+    TwoByteWord,
+    Block,
+    KiloByte,
+    MegaByte,
+    GigaByte,
+}
+
+impl Unit {
+    fn from_string(s: &str) -> Result<Unit, Box<Error>> {
+        Ok(match s {
+            "c" => Unit::Byte,
+            "w" => Unit::TwoByteWord,
+            "" | "b" => Unit::Block,
+            "k" => Unit::KiloByte,
+            "M" => Unit::MegaByte,
+            "G" => Unit::GigaByte,
+            _ => {
+                return Err(From::from(format!("Invalid suffix {} for -size. Only allowed \
+                                               values are <nothing>, b, c, w, k, M or G",
+                                              s)));
+            }
+        })
+    }
+}
+
+fn byte_size_to_unit_size(unit: Unit, byte_size: u64) -> u64 {
+    // Short circuit (to avoid a overflow error when subtracting 1 later on)
+    if byte_size == 0 {
+        return 0;
+    }
+    let divisor = match unit {
+        Unit::Byte => 1,
+        Unit::TwoByteWord => 2,
+        Unit::Block => 512,
+        Unit::KiloByte => 1024,
+        Unit::MegaByte => 1024 * 1024,
+        Unit::GigaByte => 1024 * 1024 * 1024,
+    };
+    // Skip pointless arithmetic.
+    if divisor == 1 {
+        return byte_size;
+    }
+    // We want to round up (e.g. 1 byte - 1024 bytes = 1k.
```

Comment:
The overflow can be solved by using .saturating_add() rather than simply +. It's a good point, though, you want to be very conscious of overflow behavior.

Because the divisor is dynamic, I wouldn't expect the optimization.

I don't have a strong feeling about any of this, just offering alternatives.